xbmc/filesystem/File.cpp

```diff
@@ -254,6 +254,19 @@ bool CFile::Open(const std::string& strFileName, const unsigned int flags)
 
 bool CFile::Open(const CURL& file, const unsigned int flags)
 {
+  if (m_pFile)
+  {
+    if ((flags & READ_REOPEN) == 0)
+    {
+      CLog::Log(LOGERROR, "File::Open - already open: %s", file.GetRedacted().c_str());
+      throw;
```

Comment:
throw will likely kill the app because there is no catch. please drop throw. a bid disadvantage of current c++ standard is that there is no way to enforce a catch statement.